mec/pkg/model/types_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"github.com/google/go-cmp/cmp"
+
+	"testing"
+)
+
+func TestStringToImageRef(t *testing.T) {
+	testCases := []struct {
+		input  string
+		output *ImageRef
+	}{
+		{
+			input: "registry.io/corp/repo:latest",
+			output: &ImageRef{
+				Repository: "repo",
+				Hub:        "registry.io/corp",
+				Tag:        "latest",
+			},
+		},
+		{
+			input: "registry.io/corp/repo@sha256:41af286dc0b172ed2f1ca934fd2278de4a1192302ffa07087cea2682e7d372e3",
+			output: &ImageRef{
+				Repository: "repo",
+				Hub:        "registry.io/corp",
+				SHA256:     "41af286dc0b172ed2f1ca934fd2278de4a1192302ffa07087cea2682e7d372e3",
+			},
+		},
+		{
+			input:  "registry.io/corp/repo@sha256:af286dc0b172ed2f1ca934fd2278de4a1192302ffa07087cea2682e7d372e3",
+			output: nil,
```

Comment:
if that function returned an error, you could expect an error here and in the others